src/lib/storage/table.cpp

```diff
@@ -141,8 +142,12 @@ void Table::compress_chunk(ChunkID chunk_id) {
   chunk_access_mutex.unlock();
 }
 
-void emplace_chunk(Chunk chunk) {
-  // Implementation goes here
+void Table::emplace_chunk(Chunk chunk) {
+  if (_chunks.size() == 1 && _chunks[0]->size()) {
+    _chunks[0] = std::shared_ptr<Chunk>(&chunk);
```

Comment:
here you create a shared pointer for a raw pointer to a local variable chunk. The local variable will automatically be destructed as soon as the function returns, leaving a shared pointer to a location on stack that it assumes still is is valid object but isn't